### PR TITLE
feat(nostr): support new key format

### PR DESCRIPTION
**Pull request by fczuardi**
### Describe the changes you have made in this PR

This is my attempt to address https://github.com/getAlby/lightning-browser-extension/issues/1920 the field for Nostr private keys on the settings page of the browser extension expects that the user paste a key in "hex" format. However clients like astral.ninja have adopted a new format that uses bech32 with a nsec1 prefix (see [NIP-19](https://github.com/nostr-protocol/nips/blob/master/19.md)).

This patch compares the beginning of the provided Nostr private key with the known prefixes in the NIP to check if a conversion is needed, and if so, transforms the input from bech32 to hex.

### Link this PR to an issue

Fixes _#1920_

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### How has this been tested?

I havent tested it extensively, I have compared the expected output using console log (refreshing the settings also works). And I have tried my personal `nsec` key with astral.ninja, it did seem to have signed and published new events.

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [ ] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Review by reneaaron — APPROVED** [review]
@fczuardi Thanks for your PR! Just tested it and worked fine. 👌

nit: Should we maybe add a hint that private keys are always stored in HEX format (users could get confused why their private keys do not match when they revisit the settings page)

tACK

**Comment by fczuardi** [comment]
It also bothers me a little that the settings page will display a different key when the user returns or refreshes the page. Maybe we should always display the bech32 version to the user, as the purpose of the "nsec" prefix is to prevent accidents like people pasting the private key on places expecting the public one, and in hex format this distinction is visually difficult.

**Comment by bumi** [comment]
with more interest in nostr maybe we can move this nostr things out of the settings and labs section and give it's own section (and support keys per account)

**Comment by reneaaron** [comment]
I agree, however that would give us the same problem with the value changing (but the other way around). 

Another idea would be to render the public key (npub...) below? I feel like most users will recognize their public key which would give them some indication that they've imported their key successfully.

File: src/common/lib/nostr.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { bech32Decode } from "../utils/helpers";
+
+const nostr = {
+  normalizeHex(str: string) {
+    const NIP19Prefixes = ["npub", "nsec", "note"];
```
Comment (bumi) [inline src/common/lib/nostr.ts:5]: are those always lowercase?

File: src/common/lib/nostr.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { bech32Decode } from "../utils/helpers";
```
Comment (bumi) [inline src/common/lib/nostr.ts:1]: there is a nostr/helpers.ts file - does this one also fit there?

File: src/common/lib/nostr.ts (hunk 1 above)
Comment (pqingshuang) [reply src/common/lib/nostr.ts:5]: this so ~

**Pull request merged**